2-collectors/scala-stream-collector/core/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/metrics/MetricsService.scala

```diff
@@ -0,0 +1,68 @@
+package com.snowplowanalytics.snowplow.collectors.scalastream.metrics
+
+import java.time.Duration
+
+import akka.http.scaladsl.model.{HttpMethod, StatusCode, Uri}
+import com.snowplowanalytics.snowplow.collectors.scalastream.metrics.PrometheusMetricsService.Metrics._
+import com.snowplowanalytics.snowplow.collectors.scalastream.metrics.PrometheusMetricsService.NanosecondsInSecond
+import com.snowplowanalytics.snowplow.collectors.scalastream.model.MetricsConfig
+import io.prometheus.client.exporter.common.TextFormat
+import io.prometheus.client.{CollectorRegistry, Counter, Gauge, Histogram}
+import org.apache.commons.io.output.StringBuilderWriter
+
+trait MetricsService {
+
+  def observeRequest(method: HttpMethod, uri: Uri, status: StatusCode, duration: Duration): Unit
+
+  def report(): String
+
+}
+
+class PrometheusMetricsService(metricsConfig: MetricsConfig) extends MetricsService {
+
+  private val registry = new CollectorRegistry
+  private val requestCounter: Counter =
+    Counter.build(HttpRequestCount, HttpRequestCountHelp).labelNames(Labels: _*).register(registry)
+  private val requestDurationHistogramBuilder = Histogram.build(HttpRequestDuration, HttpRequestDurationHelp).labelNames(Labels: _*)
+  private val requestDurationHistogram: Histogram =
+    metricsConfig.durationBuckets
+      .map(buckets => requestDurationHistogramBuilder.buckets(buckets: _*).register(registry))
+      .getOrElse(requestDurationHistogramBuilder.register(registry))
+
+  private val applicationVersionGauge = Gauge.build("service_version", "Java, scala versions and runnable name")
+    .labelNames("java_version", "scala_version", "runnable")
+    .register(registry)
+
+  applicationVersionGauge.labels(StaticMetrics.javaVersion, StaticMetrics.scalaVersion, StaticMetrics.runnableName).set(1)
+
+  override def observeRequest(method: HttpMethod, uri: Uri, status: StatusCode, duration: Duration): Unit = {
+    val path = uri.path.toString
+    val methodValue = method.value
+    val code = status.intValue().toString
+    requestDurationHistogram.labels(path, methodValue, code).observe(duration.toNanos / NanosecondsInSecond)
```

Comment:
why not just `toSeconds` and getting rid of the division?

Reply:
There is no public `toSeconds` and nanos give the best precision after division

Reply:
The problem is that I don't think neither nanos nor seconds are useful as a request duration unit of measurement. In my mind milliseconds would definitely be better suited, wdyt?

Reply:
Regarding to the best practices guidelines from prometheus, they encourage to use base metrics like seconds as a name for the metric. The value for milliseconds would then be a fraction of it. See here: https://prometheus.io/docs/practices/naming/#base-units